workflow/controller/operator.go

```diff
@@ -1576,3 +1610,66 @@ func expandSequence(seq *wfv1.Sequence) ([]wfv1.Item, error) {
 	}
 	return items, nil
 }
+
+// getSize return the entire workflow json string size
+func (woc *wfOperationCtx) getSize() int {
+	nodeContent, err := json.Marshal(woc.wf)
+	if err != nil {
+		return -1
+	}
+
+	compressNodeSize := len(woc.wf.Status.CompressedNodes)
+
+	if compressNodeSize > 0 {
+		nodeStatus, err := json.Marshal(woc.wf.Status.Nodes)
+		if err != nil {
+			return -1
+		}
+		return len(nodeContent) - len(nodeStatus)
+	}
+	return len(nodeContent)
+}
+
+//checkAndCompress will check the workflow size and compress node status if total workflow size is more than maxWorkflowSize.
+//The compressed content will be assign to compressedNodes element and clear the nodestatus map.
+func (woc *wfOperationCtx) checkAndCompress() error {
+
+	if woc.wf.Status.CompressedNodes != "" || (woc.wf.Status.CompressedNodes == "" && woc.getSize() >= maxWorkflowSize) {
+
+		nodeContent, err := json.Marshal(woc.wf.Status.Nodes)
+		if err != nil {
+			return errors.InternalWrapError(err)
+		}
+		buff := string(nodeContent)
+		woc.wf.Status.CompressedNodes = file.CompressEncodeString(buff)
+
+	}
+	if woc.wf.Status.CompressedNodes != "" && woc.getSize() >= maxWorkflowSize {
+		return errors.InternalError(fmt.Sprintf("Workflow is longer than maximum allowed size. Size=%d", woc.getSize()))
+	}
+	return nil
+}
+
+func (woc *wfOperationCtx) clearNodeStatusMap() {
```

Comment:
I think this is more efficient way to do it:

```
status := woc.wf.Status
status.Nodes = nil
woc.wf.Status = status
```

Reply:
updated

Reply:
wait, why isn't this just `woc.wf.Status.Nodes = nil`?

Reply:
Didn't see this question answered in latest version. Am I missing something or can this be simplified further to `woc.wf.Status.Nodes = nil`?

Reply:
updated